{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "dict_matching_res = json.load(open(\"../datasets/covid/dict-matching-res.txt\", 'r'))\n",
    "labels = json.load(open(\"../datasets/covid/labels.txt\", 'r'))\n",
    "labels = set(labels[\"LABEL\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TRAIN_DATA = [(text, {\"entities\": [(start, end, label), ...]})]\n",
    "\n",
    "def generate_entities(doc, title_offset, abs_offset):\n",
    "    entities = doc['entities']\n",
    "    entities_dict = {}\n",
    "    entities_list = []\n",
    "    for ent in entities:\n",
    "        if ent['type'] in labels:\n",
    "            start = ent['start']\n",
    "            end = ent['end']\n",
    "            label = ent['type']\n",
    "            entities_list.append((start, end, label))\n",
    "#             if end <= title_offset:\n",
    "#                 entities_list.append((start, end, label))\n",
    "#             elif start >= title_offset and end < abs_offset:\n",
    "#                 entities_list.append((start, end, label))\n",
    "#             elif start >= abs_offset:\n",
    "#                 entities_list.append((start+2, end+2, label))\n",
    "    entities_dict['entities'] = entities_list\n",
    "    return entities_dict\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_one_doc(doc):\n",
    "    title_offset = len(doc['title'])\n",
    "    abs_offset = len(doc['abstract']) + title_offset\n",
    "    text = doc['title'] + ' ' + doc['abstract'] + ' ' + doc['body']\n",
    "    entities_dict = generate_entities(doc, title_offset, abs_offset)\n",
    "    return (text, entities_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_training_data(docs):\n",
    "    res = []\n",
    "    for doc in docs:\n",
    "        res.append(generate_one_doc(doc))\n",
    "    return res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(\"The response of Milan's Emergency Medical System to the COVID-19 outbreak in Italy The number of people infected with severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), the virus causing coronavirus disease 2019 (COVID-19), is dramatically increasing worldwide.The first person-to-person transmission in Italy was reported on Feb 21, 2020, and led to an infection chain that represents the largest COVID-19 outbreak outside Asia to date. Here we document the response of the Emergency Medical System (EMS) of the metropolitan area of Milan, Italy, to the COVID-19 outbreak.On Jan 30, 2020, WHO declared the COVID-19 outbreak a public health emergency of international concern.2 Since then, the Italian Government has implemented extraordinary measures to restrict viral spread, including interruptions of air traffic from China, organised repatriation flights and quarantines for Italian travellers in China, and strict controls at international airports' arrival terminals. Local medical authorities adopted specific WHO recommendations to identify and isolate suspected cases of COVID-19.Such recommendations were addressed to patients presenting with respiratory symptoms and who had travelled to an endemic area in the previous 14 days or who had worked in the health-care sector, having been in close contact with patients with severe respiratory disease with unknown aetiology. Suspected cases were transferred to preselected hospital facilities where the SARS-CoV-2 test was available and infectious disease units were ready for isolation of confirmed cases. \",\n",
       " {'entities': [(56, 64, 'CORONAVIRUS'),\n",
       "   (77, 82, 'GPE'),\n",
       "   (97, 103, 'ORGANISM'),\n",
       "   (118, 163, 'CORONAVIRUS'),\n",
       "   (167, 177, 'CORONAVIRUS'),\n",
       "   (198, 209, 'CORONAVIRUS'),\n",
       "   (218, 222, 'DATE'),\n",
       "   (224, 232, 'CORONAVIRUS'),\n",
       "   (276, 281, 'ORDINAL'),\n",
       "   (315, 320, 'GPE'),\n",
       "   (337, 349, 'DATE'),\n",
       "   (409, 417, 'CORONAVIRUS'),\n",
       "   (512, 515, 'ORG'),\n",
       "   (552, 557, 'GPE'),\n",
       "   (566, 574, 'CORONAVIRUS'),\n",
       "   (587, 599, 'DATE'),\n",
       "   (618, 626, 'CORONAVIRUS'),\n",
       "   (705, 712, 'NORP'),\n",
       "   (816, 827, 'ORG'),\n",
       "   (833, 838, 'CELL'),\n",
       "   (891, 898, 'NORP'),\n",
       "   (899, 909, 'ORGANISM'),\n",
       "   (913, 918, 'CELL'),\n",
       "   (943, 965, 'ORG'),\n",
       "   (975, 984, 'TISSUE'),\n",
       "   (1065, 1072, 'CHEMICAL'),\n",
       "   (1092, 1100, 'CORONAVIRUS'),\n",
       "   (1230, 1250, 'DATE'),\n",
       "   (1311, 1324, 'CHEMICAL'),\n",
       "   (1473, 1483, 'CORONAVIRUS')]})"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = generate_one_doc(dict_matching_res[528])\n",
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# [inclusive, exlusive]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'COVID-19'"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d[0][56:64]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "82"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(dict_matching_res[528]['title'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': 528,\n",
       " 'source': 'CZI',\n",
       " 'doi': '10.1016/S0140-6736(20)30493-1',\n",
       " 'pmcid': '',\n",
       " 'pubmed_id': '',\n",
       " 'publish_time': 0,\n",
       " 'authors': 'Spina, Stefano; Marrazzo, Francesco; Migliari, Maurizio; Stucchi, Riccardo; Sforza, Alessandra; Fumagalli, Roberto',\n",
       " 'journal': 'The Lancet',\n",
       " 'title': \"The response of Milan's Emergency Medical System to the COVID-19 outbreak in Italy\",\n",
       " 'abstract': \"The number of people infected with severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), the virus causing coronavirus disease 2019 (COVID-19), is dramatically increasing worldwide.The first person-to-person transmission in Italy was reported on Feb 21, 2020, and led to an infection chain that represents the largest COVID-19 outbreak outside Asia to date. Here we document the response of the Emergency Medical System (EMS) of the metropolitan area of Milan, Italy, to the COVID-19 outbreak.On Jan 30, 2020, WHO declared the COVID-19 outbreak a public health emergency of international concern.2 Since then, the Italian Government has implemented extraordinary measures to restrict viral spread, including interruptions of air traffic from China, organised repatriation flights and quarantines for Italian travellers in China, and strict controls at international airports' arrival terminals. Local medical authorities adopted specific WHO recommendations to identify and isolate suspected cases of COVID-19.Such recommendations were addressed to patients presenting with respiratory symptoms and who had travelled to an endemic area in the previous 14 days or who had worked in the health-care sector, having been in close contact with patients with severe respiratory disease with unknown aetiology. Suspected cases were transferred to preselected hospital facilities where the SARS-CoV-2 test was available and infectious disease units were ready for isolation of confirmed cases.\",\n",
       " 'body': '',\n",
       " 'entities': [{'text': \"Milan's Emergency Medical System\",\n",
       "   'start': 16,\n",
       "   'end': 48,\n",
       "   'type': 'GENE_OR_GENOME'},\n",
       "  {'text': 'COVID-19', 'start': 56, 'end': 64, 'type': 'CORONAVIRUS'},\n",
       "  {'text': 'Italy', 'start': 77, 'end': 82, 'type': 'GPE'},\n",
       "  {'text': 'people', 'start': 97, 'end': 103, 'type': 'ORGANISM'},\n",
       "  {'text': 'severe acute respiratory syndrome coronavirus',\n",
       "   'start': 118,\n",
       "   'end': 163,\n",
       "   'type': 'CORONAVIRUS'},\n",
       "  {'text': 'SARS-CoV-2', 'start': 167, 'end': 177, 'type': 'CORONAVIRUS'},\n",
       "  {'text': 'coronavirus', 'start': 198, 'end': 209, 'type': 'CORONAVIRUS'},\n",
       "  {'text': '2019', 'start': 218, 'end': 222, 'type': 'DATE'},\n",
       "  {'text': 'COVID-19', 'start': 224, 'end': 232, 'type': 'CORONAVIRUS'},\n",
       "  {'text': 'first', 'start': 276, 'end': 281, 'type': 'ORDINAL'},\n",
       "  {'text': 'Italy', 'start': 315, 'end': 320, 'type': 'GPE'},\n",
       "  {'text': 'Feb 21, 2020', 'start': 337, 'end': 349, 'type': 'DATE'},\n",
       "  {'text': 'infection',\n",
       "   'start': 365,\n",
       "   'end': 374,\n",
       "   'type': 'DISEASE_OR_SYNDROME'},\n",
       "  {'text': 'COVID-19', 'start': 409, 'end': 417, 'type': 'CORONAVIRUS'},\n",
       "  {'text': 'Asia', 'start': 435, 'end': 439, 'type': 'DISEASE_OR_SYNDROME'},\n",
       "  {'text': 'EMS', 'start': 512, 'end': 515, 'type': 'ORG'},\n",
       "  {'text': 'metropolitan area',\n",
       "   'start': 524,\n",
       "   'end': 541,\n",
       "   'type': 'GENE_OR_GENOME'},\n",
       "  {'text': 'Milan', 'start': 545, 'end': 550, 'type': 'GENE_OR_GENOME'},\n",
       "  {'text': 'Italy', 'start': 552, 'end': 557, 'type': 'GPE'},\n",
       "  {'text': 'COVID-19', 'start': 566, 'end': 574, 'type': 'CORONAVIRUS'},\n",
       "  {'text': 'Jan 30, 2020', 'start': 587, 'end': 599, 'type': 'DATE'},\n",
       "  {'text': 'COVID-19', 'start': 618, 'end': 626, 'type': 'CORONAVIRUS'},\n",
       "  {'text': 'public health emergency of international concern',\n",
       "   'start': 638,\n",
       "   'end': 686,\n",
       "   'type': 'DISEASE_OR_SYNDROME'},\n",
       "  {'text': 'Italian', 'start': 705, 'end': 712, 'type': 'NORP'},\n",
       "  {'text': 'air traffic', 'start': 816, 'end': 827, 'type': 'ORG'},\n",
       "  {'text': 'China', 'start': 833, 'end': 838, 'type': 'CELL'},\n",
       "  {'text': 'quarantines',\n",
       "   'start': 875,\n",
       "   'end': 886,\n",
       "   'type': 'THERAPEUTIC_OR_PREVENTIVE_PROCEDURE'},\n",
       "  {'text': 'Italian', 'start': 891, 'end': 898, 'type': 'NORP'},\n",
       "  {'text': 'travellers', 'start': 899, 'end': 909, 'type': 'ORGANISM'},\n",
       "  {'text': 'China', 'start': 913, 'end': 918, 'type': 'CELL'},\n",
       "  {'text': 'international airports', 'start': 943, 'end': 965, 'type': 'ORG'},\n",
       "  {'text': 'terminals', 'start': 975, 'end': 984, 'type': 'TISSUE'},\n",
       "  {'text': 'authorities',\n",
       "   'start': 1000,\n",
       "   'end': 1011,\n",
       "   'type': 'SOCIAL_BEHAVIOR'},\n",
       "  {'text': 'isolate', 'start': 1065, 'end': 1072, 'type': 'CHEMICAL'},\n",
       "  {'text': 'suspected cases',\n",
       "   'start': 1073,\n",
       "   'end': 1088,\n",
       "   'type': 'GENE_OR_GENOME'},\n",
       "  {'text': 'COVID-19', 'start': 1092, 'end': 1100, 'type': 'CORONAVIRUS'},\n",
       "  {'text': 'patients', 'start': 1140, 'end': 1148, 'type': 'GROUP'},\n",
       "  {'text': 'respiratory symptoms',\n",
       "   'start': 1165,\n",
       "   'end': 1185,\n",
       "   'type': 'SIGN_OR_SYMPTOM'},\n",
       "  {'text': 'the previous 14 days', 'start': 1230, 'end': 1250, 'type': 'DATE'},\n",
       "  {'text': 'health-care',\n",
       "   'start': 1276,\n",
       "   'end': 1287,\n",
       "   'type': 'GENE_OR_GENOME'},\n",
       "  {'text': 'close contact', 'start': 1311, 'end': 1324, 'type': 'CHEMICAL'},\n",
       "  {'text': 'patients', 'start': 1330, 'end': 1338, 'type': 'GROUP'},\n",
       "  {'text': 'severe respiratory disease',\n",
       "   'start': 1344,\n",
       "   'end': 1370,\n",
       "   'type': 'GENE_OR_GENOME'},\n",
       "  {'text': 'unknown aetiology',\n",
       "   'start': 1376,\n",
       "   'end': 1393,\n",
       "   'type': 'GENE_OR_GENOME'},\n",
       "  {'text': 'Suspected cases',\n",
       "   'start': 1395,\n",
       "   'end': 1410,\n",
       "   'type': 'GENE_OR_GENOME'},\n",
       "  {'text': 'SARS-CoV-2', 'start': 1473, 'end': 1483, 'type': 'CORONAVIRUS'},\n",
       "  {'text': 'infectious disease',\n",
       "   'start': 1507,\n",
       "   'end': 1525,\n",
       "   'type': 'GENE_OR_GENOME'},\n",
       "  {'text': 'isolation',\n",
       "   'start': 1547,\n",
       "   'end': 1556,\n",
       "   'type': 'THERAPEUTIC_OR_PREVENTIVE_PROCEDURE'},\n",
       "  {'text': 'confirmed cases',\n",
       "   'start': 1560,\n",
       "   'end': 1575,\n",
       "   'type': 'GENE_OR_GENOME'}]}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict_matching_res[528]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "training_data = generate_training_data(dict_matching_res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(\"Angiotensin-converting enzyme 2 (ACE2) as a SARS-CoV-2 receptor: molecular mechanisms and potential therapeutic targetSARS-CoV-2 has been sequenced [3] . A phylogenetic analysis [3, 4] found a bat origin for the SARS-CoV-2. There is a diversity of possible intermediate hosts for SARS-CoV-2, including pangolins, but not mice and rats [5] . There are many similarities of SARS-CoV-2 with the original SARS-CoV. Using computer modeling, Xu et al. [6] found that the spike proteins of SARS-CoV-2 and SARS-CoV have almost identical 3-D structures in the receptor-binding domain that maintains van der Waals forces. SARS-CoV spike protein has a strong binding affinity to human ACE2, based on biochemical interaction studies and crystal structure analysis [7] . SARS-CoV-2 and SARS-CoV spike proteins share 76.5% identity in amino acid sequences [6] and, importantly, the SARS-CoV-2 and SARS-CoV spike proteins have a high degree of homology [6, 7] . Wan et al. [4] reported that residue 394 (glutamine) in the SARS-CoV-2 receptor-binding domain (RBD), corresponding to residue 479 in SARS-CoV, can be recognized by the critical lysine 31 on the human ACE2 receptor [8] . Further analysis even suggested that SARS-CoV-2 recognizes human ACE2 more efficiently than SARS-CoV increasing the ability of SARS-CoV-2 to transmit from person to person [4] . Thus, the SARS-CoV-2 spike protein was predicted to also have a strong binding affinity to human ACE2. This similarity with SARS-CoV is critical because ACE2 is a functional SARS-CoV receptor in vitro [9] and in vivo [10] . It is required for host cell entry and subsequent viral replication. Overexpression of human ACE2 enhanced disease severity in a mouse model of SARS-CoV infection, demonstrating that viral entry into cells is a critical step [11] ; injecting SARS-CoV spike into mice worsened lung injury. Critically, this injury was attenuated by blocking the renin-angiotensin pathway and depended on ACE2 expression [12] . Thus, for SARS-CoV pathogenesis, ACE2 is not only the entry receptor of the virus but also protects from lung injury. We therefore previously suggested that in contrast to most other coronaviruses, SARS-CoV became highly lethal because the virus deregulates a lung protective pathway [10, 12] . Zhou et al. [13] demonstrated that overexpressing ACE2 from different species in HeLa cells with human ACE2, pig ACE2, civet ACE2 (but not mouse ACE2) allowed SARS-CoV-2 infection and replication, thereby directly showing that SARS-CoV-2 uses ACE2 as a cellular entry receptor. They further demonstrated that SARS-CoV-2 does not use other coronavirus receptors such as aminopeptidase N and dipeptidyl peptidase 4 [13] . In summary, the SARS-CoV-2 spike protein directly binds with the host cell surface ACE2 receptor facilitating virus entry and replication. A key question is why the lung appears to be the most vulnerable target organ. One reason is that the vast surface area of the lung makes the lung highly susceptible to inhaled viruses, but there is also a biological factor. Using normal lung tissue from eight adult donors, Zhao et al. [14] demonstrated that 83% of ACE2-expressing cells were alveolar epithelial type II cells (AECII), suggesting that these cells can serve as a reservoir for viral invasion. In addition, gene ontology enrichment analysis showed that the ACE2-expressing AECII have high levels of multiple viral process-related genes, including regulatory genes for viral processes, viral life cycle, viral assembly, and viral genome replication [14] , suggesting that the ACE2-expressing AECII facilitate coronaviral replication in the lung. Expression of the ACE2 receptor is also found in many extrapulmonary tissues including heart, kidney, endothelium, and intestine [15] [16] [17] [18] [19] . Importantly, ACE2 is highly expressed on the luminal surface of intestinal epithelial cells, functioning as a co-receptor for nutrient uptake, in particular for amino acid resorption from food [20] . We therefore predict that the intestine might also be a major entry site for SARS-CoV-2 and that the infection might have been initiated by eating food from the Wuhan market, the putative site of the outbreak. Whether SARS-CoV-2 can indeed infect the human gut epithelium has important implications for fecal-oral transmission and containment of viral spread. ACE2 tissue distribution in other organs could explain the multi-organ dysfunction observed in patients [21] [22] [23] . Of note, however, according to the Centers for Disease Control and Prevention [24], whether a person can get COVID-19 by touching surfaces or objects that have virus on them and then touching mucus membranes is yet to be confirmed. There are several potential therapeutic approaches (Fig. 1) . Development of a spike1 subunit protein-based vaccine may rely on the fact that ACE2 is the SARS-CoV-2 receptor. Cell lines that facilitate viral replication in the presence of ACE2 may be most efficient in large-scale vaccine production. 2. Inhibition of transmembrane protease serine 2 (TMPRSS2) activity. Hoffman et al. [25] recently demonstrated that initial spike protein priming by transmembrane protease serine 2 (TMPRSS2) is essential for entry and viral spread of SARS-CoV-2 through interaction with the ACE2 receptor [26, 27] . The serine protease inhibitor camostat mesylate, approved in Japan to treat unrelated diseases, has been shown to block TMPRSS2 activity [28, 29] and is thus an interesting candidate. The interaction sites between ACE2 and SARS-CoV have been identified at the atomic level and from studies to date should also hold true for interactions between ACE2 and SARS-CoV-2. Thus, one could target this interaction site with antibodies or small molecules. Kuba et al. [10] demonstrated in mice that SARS-CoV downregulates ACE2 protein (but not ACE) by binding its spike protein, contributing to severe lung injury. This suggests that excessive ACE2 may competitively bind with SARS-CoV-2 not only to neutralize the virus but also rescue cellular ACE2 activity which negatively regulates the renin-angiotensin system (RAS) to protect the lung from injury [12, 30] . Indeed, enhanced ACE activity and decreased ACE2 availability contribute to lung injury during acid-and ventilator-induced lung injury [12, 31, 32] . Thus, treatment with a soluble form of ACE2 itself may exert dual functions: (1) slow viral entry into cells and hence viral spread [7, 9] and (2) protect the lung from injury [10, 12, 31, 32] . Notably, a recombinant human ACE2 (rhACE2; APN01, GSK2586881) has been found to be safe, with no negative hemodynamic effects in healthy volunteers and in a small cohort of patients with ARDS [33] [34] [35] . The administration of APN01 rapidly decreased levels of its proteolytic target peptide angiotensin II, with a trend to lower plasma IL-6 concentrations. Our previous work on SARS-CoV pathogenesis makes ACE2 a rational and scientifically validated therapeutic target for the current COVID-19 pandemic. The availability of recombinant ACE2 was the impetus to assemble a multinational team of intensivists, scientists, and biotech to rapidly initiate a pilot trial of rhACE2 in patients with severe COVID-19 (Clinicaltrials.gov #NCT04287686).  Josef Penninger is the founder and a shareholder of Apeiron, the company that makes rhACE2. Arthur Slutsky has been a paid consultant for Apeiron. No other conflicts of interested have been reported. This article is licensed under a Creative Commons Attribution-NonCommercial 4.0 International License, which permits any non-commercial use, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the original author(s) and the source, provide a link to the Creative Commons licence, and indicate if changes were made. The images or other third party material in this article are included in the article's Creative Commons licence, unless indicated otherwise in a credit line to the material. If material is not included in the article's Creative Commons licence and your intended use is not permitted by statutory regulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder.To view a copy of this licence, visit http://creat iveco mmons .org/licen ses/by-nc/4.0/. Potential approaches to address ACE2-mediated COVID-19 following SARS-CoV-2 infection. The finding that SARS-CoV-2 and SARS-CoV use the ACE2 receptor for cell entry has important implications for understanding SARS-CoV-2 transmissibility and pathogenesis. SARS-CoV and likely SARS-CoV-2 lead to downregulation of the ACE2 receptor, but not ACE, through binding of the spike protein with ACE2. This leads to viral entry and replication, as well as severe lung injury. Potential therapeutic approaches include a SARS-CoV-2 spike protein-based vaccine; a transmembrane protease serine 2 (TMPRSS2) inhibitor to block the priming of the spike protein; blocking the surface ACE2 receptor by using anti-ACE2 antibody or peptides; and a soluble form of ACE2 which should slow viral entry into cells through competitively binding with SARS-CoV-2 and hence decrease viral spread as well as protecting the lung from injury through its unique enzymatic function. MasR-mitochondrial assembly receptor, AT1R-Ang II type 1 receptor \",\n",
       " {'entities': [(44, 54, 'CORONAVIRUS'),\n",
       "   (55, 63, 'CHEMICAL'),\n",
       "   (120, 130, 'CORONAVIRUS'),\n",
       "   (195, 198, 'WILDLIFE'),\n",
       "   (214, 224, 'CORONAVIRUS'),\n",
       "   (259, 277, 'NORP'),\n",
       "   (282, 292, 'CORONAVIRUS'),\n",
       "   (304, 313, 'WILDLIFE'),\n",
       "   (323, 327, 'ORGANISM'),\n",
       "   (374, 384, 'CORONAVIRUS'),\n",
       "   (403, 407, 'CORONAVIRUS'),\n",
       "   (408, 411, 'CORONAVIRUS'),\n",
       "   (438, 446, 'PERSON'),\n",
       "   (485, 495, 'CORONAVIRUS'),\n",
       "   (500, 504, 'CORONAVIRUS'),\n",
       "   (505, 508, 'CORONAVIRUS'),\n",
       "   (614, 636, 'CHEMICAL'),\n",
       "   (670, 680, 'ORGANISM'),\n",
       "   (760, 770, 'CORONAVIRUS'),\n",
       "   (775, 779, 'CORONAVIRUS'),\n",
       "   (780, 783, 'CORONAVIRUS'),\n",
       "   (805, 807, 'DATE'),\n",
       "   (823, 833, 'CHEMICAL'),\n",
       "   (870, 880, 'CORONAVIRUS'),\n",
       "   (885, 889, 'CORONAVIRUS'),\n",
       "   (890, 893, 'CORONAVIRUS'),\n",
       "   (949, 958, 'PERSON'),\n",
       "   (986, 989, 'CARDINAL'),\n",
       "   (991, 1000, 'CHEMICAL'),\n",
       "   (1009, 1019, 'CORONAVIRUS'),\n",
       "   (1076, 1079, 'CARDINAL'),\n",
       "   (1083, 1087, 'CORONAVIRUS'),\n",
       "   (1088, 1091, 'CORONAVIRUS'),\n",
       "   (1127, 1133, 'CHEMICAL'),\n",
       "   (1207, 1217, 'CORONAVIRUS'),\n",
       "   (1229, 1239, 'ORGANISM'),\n",
       "   (1262, 1266, 'CORONAVIRUS'),\n",
       "   (1267, 1270, 'CORONAVIRUS'),\n",
       "   (1297, 1307, 'CORONAVIRUS'),\n",
       "   (1358, 1368, 'CORONAVIRUS'),\n",
       "   (1369, 1382, 'CHEMICAL'),\n",
       "   (1439, 1449, 'ORGANISM'),\n",
       "   (1472, 1476, 'CORONAVIRUS'),\n",
       "   (1477, 1480, 'CORONAVIRUS'),\n",
       "   (1522, 1526, 'CORONAVIRUS'),\n",
       "   (1527, 1530, 'CORONAVIRUS'),\n",
       "   (1531, 1539, 'CHEMICAL'),\n",
       "   (1591, 1600, 'CHEMICAL'),\n",
       "   (1659, 1669, 'ORGANISM'),\n",
       "   (1716, 1720, 'CORONAVIRUS'),\n",
       "   (1721, 1724, 'CORONAVIRUS'),\n",
       "   (1772, 1777, 'CELL'),\n",
       "   (1814, 1818, 'CORONAVIRUS'),\n",
       "   (1819, 1822, 'CORONAVIRUS'),\n",
       "   (1834, 1838, 'ORGANISM'),\n",
       "   (1922, 1933, 'CHEMICAL'),\n",
       "   (1991, 1995, 'CORONAVIRUS'),\n",
       "   (1996, 1999, 'CORONAVIRUS'),\n",
       "   (2035, 2049, 'PERSON'),\n",
       "   (2164, 2177, 'CORONAVIRUS'),\n",
       "   (2179, 2183, 'CORONAVIRUS'),\n",
       "   (2184, 2187, 'CORONAVIRUS'),\n",
       "   (2276, 2286, 'PERSON'),\n",
       "   (2373, 2383, 'ORGANISM'),\n",
       "   (2385, 2388, 'LIVESTOCK'),\n",
       "   (2415, 2425, 'ORGANISM'),\n",
       "   (2435, 2445, 'CORONAVIRUS'),\n",
       "   (2503, 2513, 'CORONAVIRUS'),\n",
       "   (2585, 2595, 'CORONAVIRUS'),\n",
       "   (2615, 2626, 'CORONAVIRUS'),\n",
       "   (2627, 2636, 'CHEMICAL'),\n",
       "   (2645, 2659, 'CHEMICAL'),\n",
       "   (2666, 2688, 'CHEMICAL'),\n",
       "   (2699, 2706, 'ORG'),\n",
       "   (2712, 2722, 'CORONAVIRUS'),\n",
       "   (2723, 2736, 'CHEMICAL'),\n",
       "   (2914, 2917, 'CARDINAL'),\n",
       "   (3004, 3019, 'CHEMICAL'),\n",
       "   (3073, 3084, 'TISSUE'),\n",
       "   (3090, 3095, 'CARDINAL'),\n",
       "   (3179, 3212, 'CELL'),\n",
       "   (3244, 3249, 'CELL'),\n",
       "   (3423, 3436, 'CHEMICAL'),\n",
       "   (3700, 3722, 'CHEMICAL'),\n",
       "   (3748, 3759, 'TISSUE'),\n",
       "   (3847, 3854, 'CHEMICAL'),\n",
       "   (3915, 3923, 'CHEMICAL'),\n",
       "   (3963, 3973, 'CHEMICAL'),\n",
       "   (4079, 4089, 'CORONAVIRUS'),\n",
       "   (4220, 4230, 'CORONAVIRUS'),\n",
       "   (4253, 4258, 'ORGANISM'),\n",
       "   (4259, 4273, 'CHEMICAL'),\n",
       "   (4467, 4479, 'DATE'),\n",
       "   (4518, 4545, 'CHEMICAL'),\n",
       "   (4592, 4600, 'CORONAVIRUS'),\n",
       "   (4817, 4830, 'CHEMICAL'),\n",
       "   (4869, 4879, 'CORONAVIRUS'),\n",
       "   (4880, 4888, 'CHEMICAL'),\n",
       "   (4890, 4900, 'CELL'),\n",
       "   (4996, 5003, 'CHEMICAL'),\n",
       "   (5140, 5153, 'CHEMICAL'),\n",
       "   (5250, 5260, 'CORONAVIRUS'),\n",
       "   (5290, 5308, 'DATE'),\n",
       "   (5309, 5311, 'DATE'),\n",
       "   (5335, 5344, 'CHEMICAL'),\n",
       "   (5345, 5362, 'CHEMICAL'),\n",
       "   (5376, 5381, 'GPE'),\n",
       "   (5457, 5459, 'DATE'),\n",
       "   (5538, 5542, 'CORONAVIRUS'),\n",
       "   (5543, 5546, 'CORONAVIRUS'),\n",
       "   (5669, 5679, 'CORONAVIRUS'),\n",
       "   (5731, 5741, 'CHEMICAL'),\n",
       "   (5762, 5772, 'PERSON'),\n",
       "   (5795, 5799, 'ORGANISM'),\n",
       "   (5805, 5809, 'CORONAVIRUS'),\n",
       "   (5810, 5813, 'CORONAVIRUS'),\n",
       "   (5870, 5883, 'CHEMICAL'),\n",
       "   (5901, 5919, 'CHEMICAL'),\n",
       "   (5983, 5993, 'CORONAVIRUS'),\n",
       "   (6043, 6051, 'CELL'),\n",
       "   (6103, 6114, 'CHEMICAL'),\n",
       "   (6165, 6167, 'DATE'),\n",
       "   (6266, 6270, 'CHEMICAL'),\n",
       "   (6307, 6313, 'DATE'),\n",
       "   (6315, 6317, 'DATE'),\n",
       "   (6424, 6429, 'CELL'),\n",
       "   (6502, 6504, 'DATE'),\n",
       "   (6506, 6508, 'DATE'),\n",
       "   (6510, 6512, 'DATE'),\n",
       "   (6539, 6549, 'ORGANISM'),\n",
       "   (6559, 6564, 'CHEMICAL'),\n",
       "   (6709, 6721, 'DATE'),\n",
       "   (6747, 6752, 'CHEMICAL'),\n",
       "   (6804, 6811, 'CHEMICAL'),\n",
       "   (6812, 6826, 'CHEMICAL'),\n",
       "   (6857, 6861, 'CHEMICAL'),\n",
       "   (6899, 6903, 'CORONAVIRUS'),\n",
       "   (6904, 6907, 'CORONAVIRUS'),\n",
       "   (7007, 7015, 'CORONAVIRUS'),\n",
       "   (7221, 7229, 'CORONAVIRUS'),\n",
       "   (7251, 7262, 'CHEMICAL'),\n",
       "   (7318, 7325, 'CHEMICAL'),\n",
       "   (7358, 7372, 'PERSON'),\n",
       "   (7404, 7411, 'CHEMICAL'),\n",
       "   (7546, 7567, 'CHEMICAL'),\n",
       "   (7867, 7872, 'ORDINAL'),\n",
       "   (7999, 8003, 'CELL'),\n",
       "   (8292, 8316, 'CHEMICAL'),\n",
       "   (8318, 8327, 'PERSON'),\n",
       "   (8390, 8398, 'CORONAVIRUS'),\n",
       "   (8409, 8419, 'CORONAVIRUS'),\n",
       "   (8448, 8458, 'CORONAVIRUS'),\n",
       "   (8463, 8467, 'CORONAVIRUS'),\n",
       "   (8468, 8471, 'CORONAVIRUS'),\n",
       "   (8498, 8502, 'CELL'),\n",
       "   (8554, 8564, 'CORONAVIRUS'),\n",
       "   (8600, 8604, 'CORONAVIRUS'),\n",
       "   (8605, 8608, 'CORONAVIRUS'),\n",
       "   (8620, 8630, 'CORONAVIRUS'),\n",
       "   (8712, 8725, 'CHEMICAL'),\n",
       "   (8791, 8809, 'CHEMICAL'),\n",
       "   (8854, 8864, 'CORONAVIRUS'),\n",
       "   (8865, 8878, 'CHEMICAL'),\n",
       "   (8879, 8892, 'CHEMICAL'),\n",
       "   (8910, 8918, 'CHEMICAL'),\n",
       "   (8919, 8927, 'CHEMICAL'),\n",
       "   (8938, 8947, 'CHEMICAL'),\n",
       "   (8976, 8989, 'CHEMICAL'),\n",
       "   (9057, 9065, 'CHEMICAL'),\n",
       "   (9129, 9134, 'CELL'),\n",
       "   (9170, 9180, 'CORONAVIRUS'),\n",
       "   (9333, 9337, 'ORG')]})"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "training_data[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"./training_data.txt\", 'w') as outfile:\n",
    "    json.dump(training_data, outfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
